src/bvals/boundary_conditions.cpp

```diff
@@ -221,14 +221,17 @@ void ProlongateGhostCells_(std::shared_ptr<MeshBlockData<Real>> &rc,
   std::shared_ptr<MeshBlock> pmb = rc->GetBlockPointer();
   auto &pmr = pmb->pmr;
 
-  for (auto cc_var : rc->GetCellVariableVector()) {
-    if (!cc_var->IsAllocated()) continue;
-    if (!(cc_var->IsSet(Metadata::Independent) || cc_var->IsSet(Metadata::FillGhost)))
-      continue;
-
-    // TODO(LFR): Is this indexing correct for 5 and 6 dimensional fields?
-    int nu = cc_var->GetDim(4) - 1;
-    pmr->ProlongateCellCenteredValues(cc_var.get(), si, ei, sj, ej, sk, ek);
+  for (auto var : rc->GetVariableVector()) {
+    if (!var->IsAllocated()) continue;
+    if (!(var->IsSet(Metadata::Independent) || var->IsSet(Metadata::FillGhost))) continue;
+
+    if (var->IsSet(Metadata::Cell)) {
+      // TODO(LFR): Is this indexing correct for 5 and 6 dimensional fields?
+      int nu = var->GetDim(4) - 1;
```

Comment:
`nu` is unused? At any rate I don't think prolongation/restriction has been tested for 5-, 6-D fields.

Reply:
yes, it is. I removed it.